packages/calypso-e2e/src/element-helper.ts

```diff
@@ -110,3 +110,19 @@ export async function reloadAndRetry(
 	}
 	return;
 }
+
+/**
+ * Gets and validates the block ID from a Locator to a parent Block element in the editor.
+ *
+ * @param {Locator} block A frame-safe Loccator to the top of a block.
+ * @returns A block ID that can be used to identify the block in the DOM later.
+ */
+export async function getIdFromBlock( block: Locator ): Promise< string > {
+	const blockId = await block.getAttribute( 'id' );
+	const blockIdRegex = /^block-[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}$/;
```

Comment:
What a nasty regex.

Reply:
Ha, it's actually just a GUID with `block-` in front of it 😆 . But, GUID regexes look like that 😬 .